include/fiction/io/read_gate_level_layout.hpp

```diff
@@ -0,0 +1,349 @@
+//
+// Created by simon on 25.09.23.
+//
+
+#ifndef FICTION_READ_GATE_LEVEL_LAYOUT_HPP
+#define FICTION_READ_GATE_LEVEL_LAYOUT_HPP
+
+#include "fiction/traits.hpp"
+#include "fiction/utils/name_utils.hpp"
+
+#include <fiction/layouts/cartesian_layout.hpp>
+#include <fiction/layouts/clocked_layout.hpp>
+#include <fiction/layouts/gate_level_layout.hpp>
+#include <fiction/layouts/tile_based_layout.hpp>
+
+#include <kitty/constructors.hpp>
+#include <tinyxml2.h>
+
+#include <algorithm>
+#include <cctype>
+#include <exception>
+#include <fstream>
+#include <istream>
+#include <sstream>
+#include <string>
+#include <string_view>
+#include <unordered_map>
+
+namespace fiction
+{
+
+/**
+ * Exception thrown when an error occurs during parsing of an gate_level file.
+ */
+class gate_level_parsing_error : public std::runtime_error
+{
+  public:
+    explicit gate_level_parsing_error(const std::string_view& msg) noexcept : std::runtime_error(msg.data()) {}
+};
+
+namespace detail
+{
+
+template <typename Lyt>
+class read_gate_level_layout_impl
+{
+  public:
+    read_gate_level_layout_impl(std::istream& s, const std::string_view& name) : lyt{}, is{s}
+    {
+        set_name(lyt, name);
+    }
+
+    read_gate_level_layout_impl(Lyt& tgt, std::istream& s) : lyt{tgt}, is{s} {}
+
+    Lyt run()
```

Comment:
There are not a lot of comments in this function. Would you mind adding some here and there?